lib/string_decoder.js

```diff
@@ -46,9 +50,12 @@ function StringDecoder(encoding) {
   this.lastChar = Buffer.allocUnsafe(nb);
 }
 
+// TODO(addaleax): This method should not accept strings as input.
```

Comment:
Maybe just leave the behavior about string inputs as it is and making it throw in another PR?